### PR TITLE
Hide option group with new OptionGroup constructor

### PR DESCRIPTION
**Pull request by miggs597**
Second PR that is needed to fix --help for SwiftPM

Related PRs:

[SwiftPM](https://github.com/apple/swift-package-manager/pull/3407)
[ArgumentParser](https://github.com/apple/swift-argument-parser/pull/300)

### Checklist
- [x] I've added at least one test that validates that my change is working, if appropriate
- [x] I've followed the code style of the rest of the project
- [x] I've read the [Contribution Guidelines](CONTRIBUTING.md)
- [x] I've updated the documentation if necessary

### CODE REVIEW

File: Sources/ArgumentParser/Parsable Properties/OptionGroup.swift [hunk 1]
```diff
@@ -31,6 +31,7 @@
 @propertyWrapper
 public struct OptionGroup<Value: ParsableArguments>: Decodable, ParsedWrapper {
   internal var _parsedValue: Parsed<Value>
+  internal var _hidden: Bool?
```
Comment (natecook1000) [inline Sources/ArgumentParser/Parsable Properties/OptionGroup.swift:34]: Let's have this be non-optional, and just default to `false`. Can you call it something like `_hiddenFromHelp`?

File: Sources/ArgumentParser/Parsable Properties/OptionGroup.swift [hunk 2]
```diff
@@ -84,7 +85,17 @@ extension OptionGroup: CustomStringConvertible {
     case .value(let v):
       return String(describing: v)
     case .definition:
+      if let hidden = _hidden {
+        return "OptionGroup(*definition*) _hidden: \(hidden)"
+      }
```
Comment (natecook1000) [inline Sources/ArgumentParser/Parsable Properties/OptionGroup.swift:90]: This works, but we don't really want to pass configuration values through this type's string representation. Instead, let's add `var _hiddenFromHelp: Bool { get }` as a protocol requirement in `ArgumentSetProvider`, here: https://github.com/apple/swift-argument-parser/blob/main/Sources/ArgumentParser/Parsable%20Types/ParsableArguments.swift#L230

You can add a default implementation that always returns `false`, and then this type will have the customized version, defined above.

**Review by natecook1000 — COMMENTED** [review]
This is a great start @miggs597! 👏🏻 Added some notes about a different way to do this that will let us avoid using the `description` property.

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by miggs597** [comment]
Are we ready to merge @natecook1000?

**Comment by natecook1000** [comment]
Looks good to me! 🎉

**Pull request merged**